src/main/java/com/google/songgame/servlets/UserServlet.java

```diff
@@ -49,9 +53,61 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     datastore.put(userEntity);
   }
 
+  @Override
+  public void doPut(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Map<String, String> roomProperties = readJSONFromRequest(request);
```

Comment:
You can now use my helper function (merged to master), should just be JSONParser.readJSONFromRequest(request)